torch/_subclasses/functional_tensor.py

```diff
@@ -158,6 +162,18 @@ def __new__(cls, elem):
         )
         torch._C._set_throw_on_mutable_data_ptr(out)
         out.elem = elem
+
+        if torch.is_inference_mode_enabled():
+            if out.is_base_tensor():
+                out._inference_mode_base = None
+                # This assumes that the FunctionalTensor does not change its storage after this point.
```

Comment:
Concretely, we are assuming that `out.elem` does not change its storage.

Reply:
I mention this because we do change out's storage after the `__new__` (on view operations, we patch the out tensor to have the same storage as the input to the view)